wxWidgets Survey (January 2024)

RedL0tus opened the pull request:
Topic Description
-----------------

- Update wxWidgets to 3.2.4, drop wxgtk, drop wxgtk2.
- Update wxlua to 3.2.0.2.
- Replace wxpython with wxpython-phoenix, update wxpython to 4.2.1.
- Update p11-kit to 0.25.3.
- Update gnutls to 3.8.2.
- Update libfilezilla to 0.45.0, update filezilla to 3.66.4.
- Update vba-m to 2.1.8.
- Update superslicer to 2.5.59.6.
- Update aegisub to 3.2.2+git20231203.
- Introduce portsmf 0+svn20101009.
- Update libsbsms to 2.3.0.
- Update audacity to 3.4.2.
- Drop PCem, orphaned.
- Rebuild vigra due to the stable loongson3 package being broken.
- Rebuild reverse dependencies.

Package(s) Affected
-------------------

- amule: 1:2.3.3-2
- audacity: 3.4.2
- 0ad: 0.0.26-2
- pcsx2: 1.6.0+git20210825-1
- vba-m: 1:2.1.8
- darkradiant: 3.8.0-1
- woeusb: 3.3.1-1
- gnuplot: 5.2.8-4
- wxlua: 1:3.2.0.2
- vigra: 1.11.1-14
- hugin: 2022.0.0-2
- libfilezilla: 0.45.0
- superslicer: 2.5.59.6
- wx-base: 2
- wxmaxima: 22.05.0-1
- boinc: 7.16.16-2
- kicad: 7.0.0-2
- aegisub: 3.2.2+git20231203
- wxwidgets: 3.2.4
- filezilla: 3.66.4
- wxpython: 4.2.1
- wxpython-phoenix: 1:0
- poedit: 3.4.1-1
- p11-kit: 0.25.3
- gnutls: 3.8.2
- portsmf: 0+svn20101009
- libsbsms: 2.3.0
- rlwrap: 0.43-3

Security Update?
----------------

No

Build Order
-----------

```
#buildit wxwidgets wxpython wxpython-phoenix wx-base p11-kit gnutls wxlua portsmf libsbsms audacity amule gnuplot kicad 0ad aegisub boinc darkradiant vigra hugin pcsx2 poedit superslicer vba-m rlwrap wxmaxima libfilezilla filezilla woeusb
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [ ] AArch64 `arm64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

RedL0tus committed:
wxwidgets: update to 3.2.4

- Drop outdated abicheck patch.
- Build wxbase as wxgtk3.
- Drop wxgtk2 as not needed.
- Remove hardcoded version strings.
- Explicitly set ABTYPE to autotools.
- Make beyond scripts more verbose.
- Use prepare scripts for generating .mo files.
- Use system pcre2 instead.
- Disable EGL wxGLCanvas implementation to workaround segfaults of
  PrusaSlicer and its forks.
- Cleanup stage2 definitions.
- Update PKGBREAK.
- Re-enable shadow build.
- Re-enable LTO on loongson3.

Co-Authored-By: Icenowy Zheng <icenowy@aosc.io>

RedL0tus committed:
wxgtk: drop, unneeded

RedL0tus committed:
wx-base: bump VER

- Drop wxgtk and wxgtk2 from deps.
- Remove build script and set ABTYPE=dummy.

RedL0tus committed:
wxpython: update to 4.2.1

- Replace legacy wxPython with wxPython's Project Phoenix.
- Add patch for fixing cython 3 compatibility.
- Explicitly set ABTYPE to self.

RedL0tus committed:
wxpython-phoenix: transitionalize (noarch)

RedL0tus committed:
wxlua: update to 3.2.0.2

- Switch to git source.
- Build against wxgtk3 instead of wxgtk2.
- Explicitly set ABTYPE=cmakeninja.
- Re-enable shadow build.

RedL0tus committed:
amule: bump REL

- Use wxgtk3 instead of wxgtk2.
- Remove wx-base from PKGDEP.
- Explicitly set ABTYPE=autotools.
- Re-enable shadow build and LTO.
- Use tmpfiles.d and sysusers.d for amuled user creation.
- Steal patch from Debian for fixing wxWidgets 3.2 support.

RedL0tus committed:
gnuplot: bump REL

- Use wxgtk3 instead of wxgtk2.
- Explicitly set ABTYPE=autotools.
- Re-enable shadow build and parallel compilation.

RedL0tus committed:
kicad: bump REL

- Replace wxgtk2 with wxgtk3 in PKGDEP.
- Replace wxpython-phoenix with wxpython in PKGDEP.
- Add patch to fix runtime error when KiCad was launched with non-ASCII
  character in $HOME.
- Add patch to fix confusing zh_CN localization string in the plugin updater
  dialog.
- Adjust CMake options, remove outdated option.

RedL0tus committed:
woeusb: update to 5.2.4

- Switch to git source.
- Explicitly set ABTYPE=self.
- Adjust PKGDEP according to upstream documentation.
- Replace all version placeholders with the actual version number.

RedL0tus committed:
0ad: bump REL

- Rebuild for wxgtk3 3.2.4.
- Use /usr/bin/wx-config-gtk3 as WX_CONFIG binary.
- Add llvm to BUILDDEP. SpiderMonkey's build system requires llvm-objdump.

RedL0tus committed:
aegisub: update to 3.2.2+git20231203

- Switch git src to https://github.com/TypesettingTools/Aegisub. Aegisub/Aegisub
  is no longer active and itself is a fork of TypesettingTools/Aegisub.
- Explicitly set ABTYPE=meson and re-enable shadow build.
- Add patch for icu 64 support.
- FIXME: Use system luajit once we enable lua 5.2 compat.
- FIXME: "terminate called without an active exception" on architectures other
  than amd64.

RedL0tus committed:
boinc: bump REL

- Rebuild for wxgtk3 3.2.4.
- Explicitly set ABTYPE=autotools.
- Remove outdated patch 0003-boinc-AM_CONDITIONAL.patch

RedL0tus committed:
darkradiant: bump REL

- Rebuild for wxgtk3 3.2.4.
- Explicitly set ABTYPE=cmakeninja.
- Switch to git SRCS.
- Set -DCMAKE_SKIP_RPATH=OFF to fix RPATH issue.

RedL0tus committed:
hugin: bump REL

- Rebuild for wxgtk3 3.2.4.
- Re-format CMAKE_AFTER as a single string.
- Explicitly set ABTYPE=cmakeninja.
- Add texlive to BUILDDEP, required for generating documentations.

RedL0tus committed:
pcsx2: bump REL

- Rebuild for wxgtk3 3.2.4.
- Add patch to fix compilation with wxwidgets 3.2.4.
- Explicitly set ABTYPE=cmakeninja.

RedL0tus committed:
poedit: bump REL

- Rebuild for wxgtk3 3.2.4.
- Remove unnecessary prepare script.
- Explicitly set ABTYPE=autotools.
- Re-enable shadow build.

RedL0tus committed:
superslicer: update to 2.5.59.6

- Add patches for fixing compilation and crash with wxgtk3 3.2.4.
- Use anitya for CHKUPDATE.
- Explicitly set ABTYPE=cmakeninja.

Co-Authored-By: Icenowy Zheng <icenowy@aosc.io>

RedL0tus committed:
vba-m: update to 2.1.8

- Remove outdated patches.
- Switch to git SRCS.
- Explicitly set ABTYPE=cmakeninja.

RedL0tus committed:
wxmaxima: bump REL

- Rebuild for wxgtk3 3.2.4.
- Remove unnecessary prepare files.
- Explicitly set ABTYPE=cmakeninja.

RedL0tus committed:
p11-kit: update to 0.25.3

- Re-enable shadow build.
- Explicitly set ABTYPE=autotools.
- FIXME: NOLTO=1
  warning: relocation against `p11_forkid' in read-only section `.text'

RedL0tus committed:
gnutls: update to 3.8.3

- Adjust configuration options. [^1]
  - Remove outdated options.
  - Explicitly enable DNSSEC/DANE support.
  - Enable tpm support on all architectures.
  - Enable kTLS.
  - Enable certificate compression with zlib, brotli and zstd.
- Adjust PKGDEP accordingly. [^2]
- Remove unnecessary '-ldl' from LDFLAGS.
- Enable libidn2 support.
- Cleanup stage2 definitions.
- Re-enable parallel compilation and shadow build.
- Update patch for ppc64el IEEE long double compatibility.
- Explicitly set ABTYPE=autotools.

[^1]: https://src.fedoraproject.org/rpms/gnutls/blob/rawhide/f/gnutls.spec
[^2]: https://gitlab.com/gnutls/gnutls/-/blob/9f83d394f28018986460e8ff1eb5d547dbc5f2db/INSTALL.md

RedL0tus committed:
libfilezilla: update to 0.45.0

- Explicitly set ABTYPE=autotools.
- Update PKGBREAK as sover changed.
- Remove USECLANG=1. FTBFS with clang on loongson3.

RedL0tus committed:
filezilla: update to 3.66.4

- Remove unnecessary prepare script.
- Remove unnecessary beyond script.
- Explicitly set ABTYPE=autotools.
- Remove USECLANG=1. FTBFS with clang on loongson3.
- Increase maximum text width on the about dialog for better usability on low
  resolution screens.

RedL0tus committed:
vigra: bump REL

- Rebuilding because several symbols are missing in the loongson3 stable
  package.
- Disable parallel compilation on ARM64 due to memory constraint.
- Explicitly set ABTYPE=cmakeninja.

MingcongBai commented:
Scripts look good, tests pending.

RedL0tus committed:
pcem: drop, orphaned

MingcongBai commented on app-utils/woeusb/autobuild/prepare:
```diff
@@ -0,0 +1,2 @@
+abinfo "Replacing version placeholders with $PKGVER..."
+find . -type f -print0 | xargs -0 sed -i "s/@@WOEUSB_VERSION@@/$PKGVER/"
```
Use absolute path here.

MingcongBai commented on app-web/amule/autobuild/overrides/usr/lib/systemd/system/amuled.service:
```diff
@@ -6,7 +6,7 @@ After=network.target
 [Service]
 User=amule
 ExecStart=/usr/bin/amuled
+NoNewPrivileges=true
```
Why?

MingcongBai commented on lang-python/wxpython/autobuild/build:
```diff
@@ -1,8 +1,5 @@
-            --with-regex=sys --with-libpng=sys --with-libxpm=sys --with-libjpeg=sys \
-            --with-libtiff=sys --enable-sdl --enable-libmspack
+abinfo "Building wxPython Phoenix ..."
+python3 ./build.py build_py --verbose --no_magic --use_syswx --gtk3 --release --prefix=/usr
```
Absolute path here.

MingcongBai commented on lang-python/wxpython/autobuild/build:
```diff
@@ -1,8 +1,5 @@
-python2 setup.py WXPORT=gtk2 UNICODE=1 build
-python2 setup.py WXPORT=gtk2 UNICODE=1 install --root="$PKGDIR"
+abinfo "Installing wxPython Phoenix ..."
+python3 ./build.py install_py --verbose --no_magic --use_syswx --gtk3 --release --prefix=/usr --destdir="$PKGDIR"
```
Ditto.

MingcongBai commented on runtime-cryptography/gnutls/spec:
```diff
@@ -1,5 +1,4 @@
-VER=3.6.15
-REL=1
+VER=3.8.2
```
Update to 3.8.3 to address a security issue.

MingcongBai commented on runtime-desktop/wxwidgets/01-wxbase/defines:
```diff
@@ -1,27 +1,33 @@
+                 --with-gtk=3"
 
 PKGREP="wxgtk-3<=3.0.2-3"
 PKGBREAK="wxgtk-3<=3.0.2-3"
```
Update?

MingcongBai commented on runtime-desktop/wxwidgets/01-wxbase/defines.stage2:
```diff
@@ -1,27 +1,32 @@
+                 --with-gtk=3"
 
 PKGREP="wxgtk-3<=3.0.2-3"
 PKGBREAK="wxgtk-3<=3.0.2-3"
```
Update?

MingcongBai commented on runtime-desktop/wxwidgets/02-wxgtk3/beyond:
```diff
@@ -0,0 +1,10 @@
+abinfo "Removing unneeded files ..."
+rm -fv "$PKGDIR"/usr/bin/wxrc*
```
Do not use -f.

MingcongBai commented on runtime-desktop/wxwidgets/02-wxgtk3/defines:
```diff
@@ -0,0 +1,33 @@
+                 --with-gtk=3"
+
+PKGREP="wxgtk-3<=3.0.2-3"
+PKGBREAK="wxgtk-3<=3.0.2-3"
```
Update?

RedL0tus committed:
portsmf: new, 0+svn20101009

- Convert CRLF line endings to LF.
- Steal patchset from Debian.

RedL0tus committed:
libsbsms: update to 2.3.0

- Use git SRCS.
- Explicitly set ABTYPE to cmakeninja. The upstream switched to CMake in this
  release.

RedL0tus committed:
audacity: update to 3.4.2

- Rebuild for wxWidgets 3.2.4.
- Adjust PKGDEP and CMAKE_AFTER according to upstream documentation [^1].
- Explicitly set ABTYPE to cmakeninja.

[^1]: https://github.com/audacity/audacity/blob/master/BUILDING.md

RedL0tus committed:
rlwrap: bump REL

- FIXME: Add /usr/bin/core_perl to PATH for finding pod2man.
